### PR TITLE
[Module Builder] Reset Slide

### PR DESCRIPTION
**Pull request by xsharonhe**
## Notion ticket link

[[Module Builder] Reset Slide](https://www.notion.so/uwblueprintexecs/da60bba45b0d434d964125f89ea73d93?v=a3b26293ac3f462eb6d52550a1a091a3&p=a5be509c6dda4441ae0ff445645bc9fd)

## Implementation description
* Currently, the trash button removes the slide completely. Instead, we want to have the trash button reset the slide.
* Additionally, reducer should not mutate state.

## Steps to test
1. Click trash button and ensure that a slide resets itself rather than deletes itself.

## What should reviewers focus on?
* Functionality of resetting
* Ensure reducer is written correctly

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by xsharonhe** [comment]
I'm not really sure if REMOVE_SLIDE reducer is needed anymore if the trash button has the reset button functionality? But wasn't too sure so I did not remove that code yet, so if I could get confirmation on that, that would be great.

**Comment by ChinemeremChigbo** [comment]
This PR Works as intended. However, do we plan to add a separate delete button for the individual slides? I feel like this should be made clear to the users (perhaps by having a separate "clear" button and "delete" button) as the trash icon is most commonly associated with deletion (and in other parts of the our app is used as such). I think we would want to have some sense of consistency across our app. Thoughts @xsharonhe @tawnyzhao ?

**Review by ChinemeremChigbo — APPROVED** [review]

File: src/pages/module-builder/Toolbar.tsx [hunk 1]
```diff
@@ -96,7 +96,7 @@ const Toolbar = ({
                     icon={<IoTrash />}
                     onClick={() => {
                         dispatch({
-                            type: ModuleActionType.REMOVE_SLIDE,
+                            type: ModuleActionType.RESET_SLIDE,
```
Comment (tawnyzhao) [inline src/pages/module-builder/Toolbar.tsx:99]: The previous behavior was correct actually, this line should be reverted. Sorry about that! You can leave the updated reset.

**Review by tawnyzhao — CHANGES_REQUESTED** [review]

File: src/pages/module-builder/Toolbar.tsx (hunk 1 above)
Comment (xsharonhe) [reply src/pages/module-builder/Toolbar.tsx:99]: Should I include another reset button to this? Or leave this to a future implementation

**Comment by xsharonhe** [comment]
> This PR Works as intended. However, do we plan to add a separate delete button for the individual slides? I feel like this should be made clear to the users (perhaps by having a separate "clear" button and "delete" button) as the trash icon is most commonly associated with deletion (and in other parts of the our app is used as such). I think we would want to have some sense of consistency across our app. Thoughts @xsharonhe @tawnyzhao ?

I think I agree with this which is why I was confused about changing the trash button from a delete function to a reset function, but it seems like in the future, we will be having two separate buttons, so I will revert this change.

**Review by tawnyzhao — APPROVED** [review]

**Pull request merged**